lua/indentmini/init.lua

```diff
@@ -33,8 +33,18 @@ local function indentline(hl_current)
     local hi_name = hl_group(indent)
 
     ctx[row] = indent
+     
+    local function indent_step()                                                                                                           
+      if vim.fn.exists('*shiftwidth') ~= 0 then
+        return vim.fn.shiftwidth()
+      elseif vim.fn.exists('+shiftwidth') ~= 0 then
+        return vim.opt_local.shiftwidth
+      end
+    end
+
+    local shift_step = indent_step()
```

Comment:
what about this ?

```lua
  local sw = fn.exists('*shiftwidth') == 1 and fn.shiftwidth() or vim.opt_local.shiftwidth
```

Reply:
there are cases where `opt_local.shiftwidth:get()` returns `0`. I have updated pr to add more checks.

Reply:
> what about this ?
> 
> ```lua
>   local sw = fn.exists('*shiftwidth') == 1 and fn.shiftwidth() or vim.opt_local.shiftwidth
> ```

looking at the check, it would be too long as a one-liner, but if that's what you really want, I could make the changes.

Reply:
thanks for change. and why we need use `tabstop` value ? tabstop is work when you press `tab` and it will expand how many spaces that you set in `tabstop`  when `expandtab` equals to true.  but the indent level still use `shiftwidth`